build/tools/ExtractRuleDescFromJson/RuleDescExtractor.cs

```diff
@@ -87,26 +88,61 @@ private void PrepareDestinationDirectory()
         }
     }
 
-    private static IEnumerable<Rule> LoadRules(string file)
+    private static IEnumerable<PluginRule> LoadRules(string file)
     {
+        var options = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
+
         var json = File.ReadAllText(file);
-        return JsonSerializer.Deserialize<Rule[]>(json)
-            ?? Array.Empty<Rule>();
+        return JsonSerializer.Deserialize<PluginRule[]>(json, options)
+            ?? Array.Empty<PluginRule>();
     }
 
-    private void ProcessRule(Rule rule)
+    private void ProcessRule(PluginRule pluginRule)
     {
         try
         {
-            var xml = EnsureHtmlIsXml(rule.Description);
-            SaveRuleFile(rule, xml);
+            var descAsxml = EnsureHtmlIsXml(pluginRule.Description);
+
+            var slvsRule = new RuleInfo(
+                pluginRule.Language ?? throw new ArgumentNullException("language"),
```

Comment:
We want to know at build time if there was a problem parsing the data extracted from the jars, so all of the data retrieval methods throw if data is missing.